Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -1070,49 +1070,164 @@ void StyleComputer::collect_animation_into(DOM::AbstractElement abstract_element
+        NonnullRefPtr<StyleValue const> const& underlying_value = computed_properties.property(target_property);
+        // 6. Let property-specific keyframes be the result of getting the set of computed keyframes for this keyframe effect.
+        Vector<Animations::KeyframeEffect::KeyFrameSet::ComputedKeyframe> property_specific_keyframes;
+        // 7. Remove any keyframes from property-specific keyframes that do not have a property value for target property.
+        for (auto& keyframe : computed_keyframes) {
+            if (keyframe.properties.contains(target_property)) {
+                property_specific_keyframes.append(keyframe);
             }
-            continue;
         }
 
-        if (resolved_end_property && !resolved_start_property)
-            resolved_start_property = property_initial_value(it.key);
-
-        if (!resolved_start_property || !resolved_end_property)
-            continue;
-
-        auto start = resolved_start_property.release_nonnull();
-        auto end = resolved_end_property.release_nonnull();
-
-        if (computed_properties.is_property_important(it.key)) {
-            continue;
+        // 8. If property-specific keyframes is empty, return underlying value.
+        if (property_specific_keyframes.size() == 0) {
+            return underlying_value;
         }
 
-        auto const& underlying_value = computed_properties.property(it.key);
-        if (auto composited_start_value = composite_value(underlying_value, start, start_composite_operation))
-            start = *composited_start_value;
-
-        if (auto composited_end_value = composite_value(underlying_value, end, end_composite_operation))
-            end = *composited_end_value;
+        int has_offset_zero = 0;
+        int has_offset_one = 0;
 
-        if (auto next_value = interpolate_property(*effect->target(), it.key, *start, *end, progress_in_keyframe, AllowDiscrete::Yes)) {
-            dbgln_if(LIBWEB_CSS_ANIMATION_DEBUG, "Interpolated value for property {} at {}: {} -> {} = {}", string_from_property_id(it.key), progress_in_keyframe, start->to_string(SerializationMode::Normal), end->to_string(SerializationMode::Normal), next_value->to_string(SerializationMode::Normal));
-            computed_properties.set_animated_property(it.key, *next_value);
+        for (auto keyframe : property_specific_keyframes) {
+            if (keyframe.keyframe_offset_value == 0) {
+                ++has_offset_zero;
+            } else if (keyframe.keyframe_offset_value == 1) {
+                ++has_offset_one;
+            }
+        }
+        // 9. If there is no keyframe in property-specific keyframes with a computed keyframe offset of 0, create a new keyframe with
+        //    a computed keyframe offset of 0, a property value set to the neutral value for composition,
+        //    and a composite operation of add, and prepend it to the beginning of property-specific keyframes.
```

Comment:
I think this and step 10 could do with another note mentioning why we're replacing instead. Maybe just "see step 5" to avoid repeating yourself too much.